### ISSUE TITLE
Android client stuck on loading Ansi Latin 1 and UTF16 .md and .txt files with German umlauts

### ISSUE THREAD

**Issue by Sharky444** [nextcloud/android#6026]
I have this text with the German umlaut Letter "Ü".

```
Über test 2

abc
```

I save this text into 3 text files as ASCII/Ansi Latin 1, as UTF8, and as UTF16.

Result in Nextcloud Android client:

The Ansi latin 1 version is stuck indefinitely on loading.
The UTF8 version can be opened and edited.
The UTF16 version is stuck indefinitely on loading.

Ansi Text is still widely used, both on Windows and Linux. Any German umlaut in such text files makes the file unloadable (even for display only) in the Android client.

Compare that to the web client, where all 3 text files can be opened and edited, although the ansi latin 1 will not display correctly (UTF8 and UTF16 will display correctly).

[Umlaut_AnsiLatin1.txt](https://github.com/nextcloud/android/files/4598693/Umlaut_AnsiLatin1.txt)
[Umlaut_UTF8.txt](https://github.com/nextcloud/android/files/4598696/Umlaut_UTF8.txt)
[Umlaut_UTF16.txt](https://github.com/nextcloud/android/files/4598697/Umlaut_UTF16.txt)

**Comment by AndyScherzinger** [comment]
cc @nextcloud/server-triage and @tobiasKaminsky

**Comment by rullzer** [comment]
@Sharky444 this is the in app editing I guess?

Does it load for you on the webui?

**Comment by Sharky444** [comment]
As stated on the end of my bug report, it does.

**Comment by tobiasKaminsky** [comment]
So you create the file, upload it via web and then?
Click on it in Android to view it?
Or download it first and then click on it?
(as this are two different systems I need to know which one is the problematic).
Please also test vice vera.

**Comment by Sharky444** [comment]
I've created the files under Windows, then the Windows Desktop client has uploaded them to the cloud. I see the files in the Android app, but only the UTF8 will open, Ansi Latin 1 version and UTF 16 version will cause loading screen indefinitely (and a message that the file takes long to load).

When I access the files using the website instead, all three versions can be displayed and edited with the web client (although the Ansi Latin version will not display the Ü char, but something different).

p.s.

Just tested the iOS app and it has the same bug.

**Comment by tobiasKaminsky** [comment]
What happens when you first download the file on Android and then open it?

@juliushaertl any idea, as we only open the link in a webview?

**Comment by Sharky444** [comment]
When I download the files first they open and display correctly.

**Comment by juliushaertl** [comment]
Yes, seems to be some issue in the text app when loading the file we don't return a proper content-type.

```
> GET /apps/text/session/fetch?documentId=1529646&sessionId=174&sessionToken=MzhrmRnq8I5NupJbSgnZo20naA4L%2FO9P7wzZt76BQjnRbJnphjze06W3CHJs1kab HTTP/2
...
< content-type: text/html; charset=UTF-8
< 
�ber test 2
```
